zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -0,0 +1,552 @@
+                OpcodeId::DUP3,
+                OpcodeId::DUP4,
+                OpcodeId::DUP5,
+                OpcodeId::DUP6,
+                OpcodeId::DUP7,
+                OpcodeId::DUP8,
+                OpcodeId::DUP9,
+                OpcodeId::DUP10,
+                OpcodeId::DUP11,
+                OpcodeId::DUP12,
+                OpcodeId::DUP13,
+                OpcodeId::DUP14,
+                OpcodeId::DUP15,
+                OpcodeId::DUP16,
+            ],
+            Self::SWAP => vec![
+                OpcodeId::SWAP1,
+                OpcodeId::SWAP2,
+                OpcodeId::SWAP3,
+                OpcodeId::SWAP4,
+                OpcodeId::SWAP5,
+                OpcodeId::SWAP6,
+                OpcodeId::SWAP7,
+                OpcodeId::SWAP8,
+                OpcodeId::SWAP9,
+                OpcodeId::SWAP10,
+                OpcodeId::SWAP11,
+                OpcodeId::SWAP12,
+                OpcodeId::SWAP13,
+                OpcodeId::SWAP14,
+                OpcodeId::SWAP15,
+                OpcodeId::SWAP16,
+            ],
+            Self::LOG => vec![
+                OpcodeId::LOG0,
+                OpcodeId::LOG1,
+                OpcodeId::LOG2,
+                OpcodeId::LOG3,
+                OpcodeId::LOG4,
+            ],
+            Self::CREATE => vec![OpcodeId::CREATE],
+            Self::CALL => vec![OpcodeId::CALL],
+            Self::CALLCODE => vec![OpcodeId::CALLCODE],
+            Self::RETURN => vec![OpcodeId::RETURN],
+            Self::DELEGATECALL => vec![OpcodeId::DELEGATECALL],
+            Self::CREATE2 => vec![OpcodeId::CREATE2],
+            Self::STATICCALL => vec![OpcodeId::STATICCALL],
+            Self::REVERT => vec![OpcodeId::REVERT],
+            Self::SELFDESTRUCT => vec![OpcodeId::SELFDESTRUCT],
+            _ => vec![],
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct StepState<F> {
+    pub(crate) execution_result: Vec<Cell<F>>,
+    pub(crate) rw_counter: Cell<F>,
+    pub(crate) call_id: Cell<F>,
+    pub(crate) is_root: Cell<F>,
+    pub(crate) is_create: Cell<F>,
+    pub(crate) opcode_source: Cell<F>,
+    pub(crate) program_counter: Cell<F>,
+    pub(crate) stack_pointer: Cell<F>,
+    pub(crate) gas_left: Cell<F>,
+    pub(crate) memory_size: Cell<F>,
+    pub(crate) state_write_counter: Cell<F>,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct StepRow<F> {
+    pub(crate) qs_byte_lookup: Cell<F>,
+    pub(crate) cells: [Cell<F>; STEP_WIDTH],
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct Step<F> {
+    pub(crate) state: StepState<F>,
+    pub(crate) rows: Vec<StepRow<F>>,
+}
+
+impl<F: FieldExt> Step<F> {
+    pub(crate) fn new(
+        meta: &mut ConstraintSystem<F>,
+        qs_byte_lookup: Column<Advice>,
+        advices: [Column<Advice>; STEP_WIDTH],
+        is_next_step: bool,
+    ) -> Self {
+        let state = {
+            let mut cells =
+                VecDeque::with_capacity(ExecutionResult::amount() + 10);
```

Comment:
why plus 10 for cells capacity ?

Reply:
For other stuff in `StepState`.

Reply:
Fixed in [`22fd394`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/22fd39411d7083466a40fe62b2622e6b192b354d).